File: Kyu4/text_align_justify.js
export function justify(text, width) {
  const words = text.split(" ");

  //   let textArr = [];
  let texts = "";

  let tempWidth = width;
  for (let i = 0; i < words.length; i++) {
    if (tempWidth - words[i].length >= 0) {
      tempWidth -= words[i].length;
      //   textArr.push(words[i]);
      texts += words[i];
    } else {
      //   textArr.push("\n");
      texts += "\n";

      tempWidth = width;
    }
    // while (tempWidth - words[i].length >= 0) {
    //   tempWidth -= words[i].length;
    //   textArr.push(words[i]);
    //   words.shift();
    //   i++;
    // }
    // textArr.push("\n");
  }

  console.log({ texts });

  //   let texts = [];
  //   for (let i = 0; i < Math.floor(text.length / width); i++) {
  //     texts.push(text.slice(i * width, width));
  //   }
  //   texts.push(text.slice(Math.floor(text.length / width) * width, text.length));
  //   console.log({ texts });

  //   return texts.join("\n");
}

File: Kyu5/text_align_justify.py
def justify(text, width):
    words = text.split(" ")
    count = 0
    i = 0
    lines = []
    lineArr = []

    # while (i < len(words)):
    #     print(i, count, words[i])
    #     count += (len(words[i]) + 1)
    #     if count > width:
    #         lines.append(lineArr)
    #         lineArr = []
    #         count = 0
    #         continue
        
    #     lineArr.append(words[i])
    #     i += 1

    # for w in words:
    #     count += len(w) + 1
    #     print(count, w)
    #     if count > width:
    #         count = 0
    #         lines.append(lineArr)
    #         lineArr = []

    #     lineArr.append(w)
    #     print("lineArr:", lineArr)

    #     if w == words[-1]: 
    #         lines.append(lineArr)

    for w in words:
        print(w, count)

        count += len(w) + 1 if w != words[-1] else len(w)

        if count <= width:
            lineArr.append(w)
            # print("lineArr:", lineArr)
        else: 
            count = len(w) + 1
            lines.append(lineArr)
            lineArr = [w]

        if w == words[-1]: 
            lines.append(lineArr)
        

    print(lines)

print(justify('123 45 6', 7)) # '123  45\n6'
